qemu/tests/qemu_guest_agent.py

```diff
@@ -582,6 +582,70 @@ def gagent_check_set_time(self, test, params, env):
             if delta > 3:
                 test.fail("The guest time can't be set from hwclock on host")
 
+    @error_context.context_aware
+    def gagent_check_time_sync(self, test, params, env):
+        """
+        Run "guest-set-time" to sync time after stop/cont vm
+
+        Steps:
+        1) start windows time service in guest and
+        change ntp server to clock.redhat.com
+        2) stop vm
+        3) wait 3 mins and resume vm
+        4) execute "guest-set-time" cmd via qga
+        5) query time offset of vm,it should be less than 3 seconds
+
+        :param test: kvm test object
+        :param params: Dictionary with the test parameters
+        :param env: Dictionary with test environment.
+        """
+        timeout = float(params.get("login_timeout", 240))
+        time_config_cmd = params["time_service_config"]
+        time_service_status_cmd = params["time_service_status_cmd"]
+        time_service_start_cmd = params["time_service_start_cmd"]
+        time_service_stop_cmd = params["time_service_stop_cmd"]
+
+        session = self.vm.wait_for_login(timeout=timeout)
+
+        error_context.context("Start windows time service.", logging.info)
+        if session.cmd_status(time_service_status_cmd):
+            session.cmd(time_service_start_cmd)
+
+        error_context.context("Config time resource and restart time"
+                              " service.", logging.info)
+        session.cmd(time_config_cmd)
+        session.cmd(time_service_stop_cmd)
+        session.cmd(time_service_start_cmd)
+
+        error_context.context("Stop the VM", logging.info)
+        self.vm.pause()
+        error_context.context("Verify the status of VM is 'paused'",
+                              logging.info)
+        self.vm.verify_status("paused")
+        session.close()
+
+        time.sleep(float(params["pause_time"]))
+        error_context.context("Resume the VM", logging.info)
+        self.vm.resume()
+        error_context.context("Verify the status of VM is 'running'",
+                              logging.info)
+        self.vm.verify_status("running")
+
+        error_context.context("Execute guest-set-time cmd.", logging.info)
+        self.gagent.set_time()
+
+        logging.info("Wait a few seconds up to a minute to check guest time.")
+        endtime = time.time() + 60
```

Comment:
@xiagao  I think endtime <30 seconds is better.   Guest time may happen time drift in 60 seconds. To avoid this,  you can set endtime <30 or <20 or <10 seconds.